### PR TITLE
BREAKING feat: use hashV1 and assimilate buildEntityAndFile from commons

### PR DESCRIPTION
**Pull request by menduz**
Based on the usage of `buildEntityAndFile` across repositories what makes more sense is to assimilate that function inside the client.

The PR original's purpose was to upgrade the hashing algorithm to `CIDv1`. The assimilation of the function was necessary.

There is a second PR in `catalyst-commons` to remove the duplicated function https://github.com/decentraland/catalyst-commons/pull/168

### CODE REVIEW

**Review by guidota — APPROVED** [review]
LGTM 👍

File: src/utils/DeploymentBuilder.ts [hunk 1]
```diff
@@ -1,18 +1,80 @@
...
+    metadata?: EntityMetadata
+  }): Promise<{ entity: Entity; entityFile: Uint8Array }> {
+    // Make sure that there is at least one pointer
+    if (pointers.length === 0) throw new Error(`All entities must have at least one pointer.`)
+
+    if (version === EntityVersion.V2) throw new Error(`V2 is not supported.`)
+
+    const entity = {
+      // default version is V3
+      version: version || EntityVersion.V3,
```
Comment (agusaldasoro) [inline src/utils/DeploymentBuilder.ts:43]: If the version is deprecated shouldn't we don't do anything about it? Also, wouldn't the client build the latest version of the entitites?

Comment (menduz) [reply src/utils/DeploymentBuilder.ts:43]: the last version is v3. V4 are not needed as of ADR51.

The migration plan would require stop making the "version" field required on the catalyst but I would implement that change later on to not block the migration to the new hashes

File: package.json [hunk 2]
```diff
@@ -43,6 +43,8 @@
   },
   "homepage": "https://github.com/decentraland/catalyst-client#readme",
   "dependencies": {
+    "@dcl/hashing": "^1.0.0",
+    "@dcl/schemas": "^3.8.0",
     "@types/form-data": "^2.5.0",
     "cookie": "^0.4.1",
     "dcl-catalyst-commons": "7.2.1",
```
[the remark this answers is not in the document]
Comment (menduz) [reply package.json:50]: catalyst-commons? I was waiting for https://github.com/decentraland/catalyst-commons/pull/168 to do it

**Pull request merged**